src/middleware/logger/index.ts

```diff
@@ -28,19 +28,22 @@ const time = (start: number) => {
 
 const colorStatus = (status: number) => {
   const colorEnabled = getColorEnabled()
-  const out: { [key: string]: string } = {
-    7: colorEnabled ? `\x1b[35m${status}\x1b[0m` : `${status}`,
-    5: colorEnabled ? `\x1b[31m${status}\x1b[0m` : `${status}`,
-    4: colorEnabled ? `\x1b[33m${status}\x1b[0m` : `${status}`,
-    3: colorEnabled ? `\x1b[36m${status}\x1b[0m` : `${status}`,
-    2: colorEnabled ? `\x1b[32m${status}\x1b[0m` : `${status}`,
-    1: colorEnabled ? `\x1b[32m${status}\x1b[0m` : `${status}`,
-    0: colorEnabled ? `\x1b[33m${status}\x1b[0m` : `${status}`,
+  if (colorEnabled) {
+    switch ((status / 100) | 0) {
+      case 5: // red = error
+        return `\x1b[31m${status}\x1b[0m`
+      case 4: // yellow = warning
+        return `\x1b[33m${status}\x1b[0m`
+      case 3: // cyan = redirect
+        return `\x1b[36m${status}\x1b[0m`
+      case 2: // green = success
+        return `\x1b[32m${status}\x1b[0m`
+    }
   }
-
-  const calculateStatus = (status / 100) | 0
-
-  return out[calculateStatus]
+  // Fallback to unsupported status code.
+  // E.g.) Bun and Deno supports new Response with 101, but Node.js does not.
+  // And those may evolve to accept more status.
+  return `${status}`
```

Comment:
Is this line tested?

Reply:
The line is covered for color-disabled case.
But not covered for currently unsupported status code, because test always fail if we test `100` or `700`.
This fallback will help when `Response` supports `100` or other in future.

Reply:
Hi @exoego, thanks for creating the pull request.

It's a fine line as to whether we need to overwrite the object and test it, but it's better if there is test, so how about testing it with the following code?

```diff
diff --git a/src/middleware/logger/index.test.ts b/src/middleware/logger/index.test.ts
index 0623fc64..7c7e4689 100644
--- a/src/middleware/logger/index.test.ts
+++ b/src/middleware/logger/index.test.ts
@@ -32,7 +32,12 @@ describe('Logger by Middleware', () => {
       return c.redirect('/empty', 301)
     })
     app.get('/server-error', (c) => {
-      return c.text('', 511)
+      const res = new Response('', { status: 511 })
+      if (c.req.query('status')) {
+        // test status code not yet supported by runtime `Response` object
+        Object.defineProperty(res, 'status', { value: parseInt(c.req.query('status') as string) })
+      }
+      return res
     })
   })
 
@@ -95,6 +100,22 @@ describe('Logger by Middleware', () => {
     expect(log.startsWith('--> GET /server-error \x1b[31m511\x1b[0m')).toBe(true)
     expect(log).toMatch(/m?s$/)
   })
+
+  it('Log status 100', async () => {
+    const res = await app.request('http://localhost/server-error?status=100')
+    expect(res).not.toBeNull()
+    expect(res.status).toBe(100)
+    expect(log.startsWith('--> GET /server-error 100')).toBe(true)
+    expect(log).toMatch(/m?s$/)
+  })
+
+  it('Log status 700', async () => {
+    const res = await app.request('http://localhost/server-error?status=700')
+    expect(res).not.toBeNull()
+    expect(res.status).toBe(700)
+    expect(log.startsWith('--> GET /server-error 700')).toBe(true)
+    expect(log).toMatch(/m?s$/)
+  })
 })
 
 describe('Logger by Middleware in NO_COLOR', () => {
```

Reply:
Oh that's hacky but nice. 
Done in https://github.com/honojs/hono/pull/3603/commits/66f35bf6626fa10118ad9df055599795d3e3a0f1

Reply:
Thank you!